org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js

```diff
@@ -22,101 +16,82 @@ function validateArgumentType(arg, argName, expectedType) {
-        return helper.setProperty({'hvac_mode' : 'off'});
-    },
-
-    getCurrentTemperature : function() {
-        console.log('getCurrentTemperature called.');
-        return helper.getProperty('ambient_temperature_c');
-    },
-
-    getHeatingSetpoint : function() {
-        console.log('getHeatingSetpoint called.');
-        return helper.getProperty('target_temperature_high_c');
-    },
-
-    setHeatingSetpoint : function(value) {
-        console.log("setHeatingSetpoint called");
-         return helper.setProperty({'target_temperature_high_c' : value});
-    },
-
-     getCoolingSetpoint : function() {
-        console.log('getCoolingSetpoint called.');
-        return helper.getProperty('target_temperature_low_c');
-    },
-
-    setCoolingSetpoint : function(value) {
-        console.log("setCoolingSetpoint called.");
-        return helper.setProperty({'target_temperature_low_c' : value});
-    },
-
-    getMode : function() {
-        console.log('getMode called.');
-        return helper.getProperty('hvac_mode', 'off');
-    },   
-
-    setMode : function(value) {
-       console.log("setMode called."); 
-       return helper.setProperty({'hvac_mode' : value});
-    },
-    
-    getAvailableModes: function(value) {
-       var deferred = q.defer();   // Take a deferral
-       deferred.resolve(['heat', 'cool', 'heat-cool','off']);
-       return deferred.promise; // return the promise
-    },
-}
\ No newline at end of file
+        validateArgumentType(device.props, 'device.props', 'object');
+
+        validateArgumentType(device.props.access_token, 'device.props.access_token', 'string');
+        validateArgumentType(device.props.id, 'device.props.id', 'string');
+
+        deviceId = device.props.id;
+
+        // Initialize Nest Helper
+        nestHelper = new NestHelper(device.props.access_token);
+        console.log('Javascript and Nest Helper initialized : ');
+    }
+
+    // exports for the OCF schema
+
+    // Queries the entire state of the thermostat
+    // and returns an object that maps to the json schema org.opent2t.sample.thermostat.superpopular
+    getThermostatResURI() {
+        return nestHelper.getDeviceDetailsAsync(deviceType, deviceId);
+    }
+
+    // Updates the current state of the thermostat with the contents of value
+    // value is an object that maps to the json schema org.opent2t.sample.thermostat.superpopular
+    //
+    // In addition, returns the updated state (see sample in RAML)
+    postThermostatResURI(value) {
+        // build the post format Nest depends on
+        var putPayload = {};
+        putPayload['target_temperature_c'] = value.targetTemperature;
+        putPayload['target_temperature_high_c'] = value.targetTemperatureHigh;
+        putPayload['target_temperature_low_c'] = value.targetTemperatureLow;
+
+        return nestHelper.putDeviceDetailsAsync(deviceType, deviceId, putPayload);
+    }
+
+    // exports for the AllJoyn schema
+    getAmbientTemperature() {
```

Comment:
At least we have automated tests for this... I also think there are two use cases in the cloud service sample @arjun-msft is writing:

1. GET or POST to /devices/<deviceId> (returns or posts a json blob that looks like our super popular thermostat schema). For this case, use the OCF methods.

2. GET or POST to /devices/<deviceId>/propertyName (returns or posts only the value of the requested property). Under the hood, this COULD call the AllJoyn methods... might be useful for callers who don't want to fetch / post the ENTIRE device object if all they want is one property.

Obviously, we are not thinking about the second one quite yet... but just to discuss I think rather than OCF/AllJoyn methods we might want to think about Object/Property methods. Food for thought.